src/history/api.py

```diff
@@ -3,16 +3,17 @@
 from datetime import datetime
 from pathlib import Path
 from typing import Union
+import re
 
 ROUNDS = 4
 INF = 2147483647
 NA = -1
 BASE_DIR = Path(Path(__file__).resolve()).parent
 
-
+YEAR_REGEX = re.compile("[0-9]{8}")
```

Comment:
E302 expected 2 blank lines, found 1 [E302]

I think it would make the most sense to move this constant right above line 30, to ensure that the function block has 2 blanklines above it. This is probably because `YEAR_REGEX` at first glance probably sounds like it should be `[0-9]{4}[ /-][0-9]{4}` if placed in the global scope, so moving it and restricting it to the context above line 30 could be better